### PR TITLE
Header & CMS site

### PR DESCRIPTION
**Pull request by Crashillo**
### What does this PR do?

1. Includes [Getafe CMS sandbox view](http://madrid.gobierto.dev/sandbox/home_cms.html)
2. Fully redesign gobierto header all around 

### Pending

- Mobile view. For the time being, is just a 1st approach. Lack of some functions (login, account e.g.) 

### Considerations
Sandbox mockups turned to use actual header & footer. Keep that in mind.

## UPDATES

- Removes private variables (font & colors) to take the default ones, or the customized in the admin pages.
- Includes, in the mockup, inline style tag to modify CMS default look.
- Added two new directives, at *CMS page* template, to make previous step work.
- Sandbox updated available [here](https://raw.githubusercontent.com/PopulateTools/gobierto/39b63e7285581fce50be95588a953d3f209ec8e5/app/views/sandbox/home_cms.html.erb)

### Considerations
CMS Articles have a wrapper, based on *purecss*,  that should disappear to make it full-width. Class is *[pure-u-md-3-4](https://github.com/PopulateTools/gobierto/blob/39b63e7285581fce50be95588a953d3f209ec8e5/app/views/gobierto_cms/templates/_page.html.erb#L22)*

### CODE REVIEW

File: app/assets/stylesheets/comp-site_header.scss [hunk 1]
```diff
@@ -1,542 +1,369 @@
...
- *
- */
-header.meta {
-  padding: 0;
-  margin: 0 0 2em 0;
-  @include screen(740) {
-    margin: 0;
+header {
+  // Header colors
+  $cBlue: #0065C8;
```
Comment (furilo) [inline app/assets/stylesheets/comp-site_header.scss]: we have the colors specified in other place

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  <% end %>
+
+  <div class="site_header">
+    <div class="column site_header_block">
+
+      <div class="site_header_logo">
+        <div class="site_header_image">
+          <%= link_to(image_tag(@site.configuration.logo_with_fallback, alt: @site.name), root_url) %>
+        </div>
+        <div class="site_header_slogan">
```
Comment (furilo) [inline app/views/layouts/_header.html.erb]: this was not meant to be converted to HTML, it was part of the logo in that specific case

File: app/assets/stylesheets/comp-site_header.scss [hunk 3]
```diff
@@ -1,542 +1,369 @@
...
+  $fBold: 700;
+
+  // Header media breakpoints, from purecss.io
+  $wSmall: 568px;
+  $wMedium: 768px;
+  $wLarge: 1024px;
+  $wExtraLarge: 1280px;
+
+  // Header Mixins
+  @mixin font($size: $f6, $color: $cBlue, $weight: $fRegular, $style: normal) {
```
Comment (furilo) [inline app/assets/stylesheets/comp-site_header.scss]: we should not have this kind of things, which are generic (used by many comps), in this specific place

File: app/assets/stylesheets/comp-cms.scss [hunk 4]
```diff
@@ -0,0 +1,130 @@
+// Colors
+$white: #ffffff;
+$black: #000000;
```
Comment (furilo) [inline app/assets/stylesheets/comp-cms.scss]: we have colors definitions in another place

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    <div class="column pure-g">
+
+      <div class="desktop_only">
+
+        <div class="pure-u-md-1-2">
+          <% if @site.configuration.links_markup.present? %>
+            <%= raw @site.configuration.links_markup %>
+          <% end %>
+
+          <%# TODO: This links are static %>
```
Comment (furilo) [inline app/views/layouts/_header.html.erb]: please review original templates, we have dynamic code for this

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+                      <%= t("layouts.application.session.sign_in") %>
+                    <% end %>
+                  </li>
+                </ul>
+              <% end %>
+
+            </div>
+          </div>
+
+          <%# TODO: This links are static %>
```
Comment (furilo) [inline app/views/layouts/_header.html.erb]: again, there is dynamic code for this already implemented. this: 

https://github.com/PopulateTools/gobierto/pull/1181/files#diff-faf6482c7d306c24c4e64dcdf191f7f4R113

File: app/assets/stylesheets/comp-site_header.scss [hunk 7]
```diff
@@ -1,542 +1,369 @@
...
+  $cLightBlue: rgba($cBlue, 0.1);
+
+  // Header weights
+  $fExtraLight: 200;
+  $fRegular: 400;
+  $fSemibold: 600;
+  $fBold: 700;
+
+  // Header media breakpoints, from purecss.io
+  $wSmall: 568px;
```
Comment (furilo) [inline app/assets/stylesheets/comp-site_header.scss:9]: this should be generic so don't belong here

File: app/assets/stylesheets/comp-site_header.scss (hunk 1 above)
Comment (furilo) [reply app/assets/stylesheets/comp-site_header.scss]: we should be using $color_main, $color_main_soft etc defined in css-conf.scss. those are the colors and we shouldn't create more (unless there is a very good reason). the screen you had as reference was a "custom theme". 

we are defining those overriding some classes to change colors, directly in the admin of each site (I thought you were already familiar with that). take a look at getafe, mataro, sant feliu... in staging/pro, in "personalizar sitio > html de cabecera". 

so: you should be using always the base colors only

**Review by furilo — CHANGES_REQUESTED** [review]

**Comment by ferblape** [comment]
Lo dejo por el momento hasta que ajustéis las cosas que faltan y estas que añado:

### Página de proceso

La navegación de proceso tiene un Breadcrumb de proceso en la `main-nav` y una navegación secundaria en la `sub-nav`. Habría que ajustar ambas:

screen shot 2017-12-01 at 10 50 02

### Altos Cargos

Dentro de Cargos, Agendas, Declaraciones o Blogs hay muy poco margen:

screen shot 2017-12-01 at 10 53 19

Cuando esté esto listo termino de arreglar los tests, porque dependen un poco en el marcado HTML.

**Comment by furilo** [comment]
Please review @ferblape 

Things that might be improved, although we may leave for another PR: 

- We should show secondary nav not only for processes, but for all participation as we have now (in the future, it may make sense to show this for all Gobierto)
- When in a Process, maybe add the .active to .main-nav-item ?
- I don't like very much the new position of the follow button. Let's leave it there but we should revisit it.
- Mobile nav is not very good. I'll open another issue.

**Pull request merged**